### PR TITLE
✨ Support build.yaml to configure

### LINKED ISSUE #114: [BUG] build.yaml configuration instead of pubspec.yaml
build.yaml intended for generators configuration.
So could be nice to use build.yaml configuration instead of pubspec.yaml configuration.

❌ **Bad**  
```dart
pubspec.yaml
```

⭕️ **Good**  
```dart
build.yaml
```

### PR DESCRIPTION
**Pull request by b2nkuu**
## What does this change?

Read config build.yaml and replacement configure after default, pubspec.yaml. 

Priority of configuration file.
High ---> Low
build.yaml -> pubspec.yaml -> default_config

Fixes #114 🎯

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [x] Make sure to open a GitHub issue as a bug/feature request before writing your code! That way we can discuss the change, evaluate designs, and agree on the general idea
  - [x] Ensure the tests (`melos run test`)
  - [x] Ensure the analyzer and formatter pass (`melos run format` to automatically apply formatting)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

**Comment by AlexV525** [comment]
Could you clarify the priority when both configs in `pubspec.yaml` and `build.yaml` exist?

**Comment by b2nkuu** [comment]
> Could you clarify the priority when both configs in `pubspec.yaml` and `build.yaml` exist?

Yep, Priority of configuration file.
High ---> Low
build.yaml -> pubspec.yaml -> default_config

**Comment by AlexV525** [comment]
> Yep, Priority of configuration file. High ---> Low build.yaml -> pubspec.yaml -> default_config

Thanks, in this case I would like to see if we can log what configuration the gen command is using, probably like: `Reading FlutterGen options from xxxx.xxx.`.

**Comment by b2nkuu** [comment]
> > Yep, Priority of configuration file. High ---> Low build.yaml -> pubspec.yaml -> default_config
> 
> Thanks, in this case I would like to see if we can log what configuration the gen command is using, probably like: `Reading FlutterGen options from xxxx.xxx.`.

Got it.
I will add command to select file to read and default at pubspec.yaml 
OK?

**Comment by AlexV525** [comment]
> Got it. I will add command to select file to read and default at pubspec.yaml OK?

We might focus on one specific feature in one request. Besides, we don't need such rich options AFAICT.

File: README.md [hunk 1]
```diff
@@ -296,6 +296,28 @@ class $AssetsImagesGen {
 }
 ```
 
+#### Support generate at **build.yaml**
+Support generates dart files based on the key **`targets.$default.builders`** and **`flutter_gen.options`** 
+of [`build.yaml`](https://github.com/dart-lang/build/blob/master/build_config/README.md).
```
[the remark this answers is not in the document]
Comment (AlexV525) [reply README.md:319]: Ah so this section should be deleted then?

File: packages/core/test_resources/build_output.yaml
Comment (AlexV525) [inline packages/core/test_resources/build_output.yaml:1]: This would produce actual data, right? If so, add the data under `test_resources/actual_data`.

**Review by AlexV525 — APPROVED** [review]
Thank you for committing to all of the suggestions!

**Pull request merged**